cli/cli.go

```diff
@@ -138,24 +158,36 @@ func (c *CLI) Stream() error {
 		}
 		// if that's all we've got for now, wait 5 seconds then try again
 		if !*resp.AdditionalDataPending || *resp.Marker == "0" {
-			time.Sleep(5 * time.Second)
+			c.waitFor(5 * time.Second)
 		}
-		oldMarker := resp.Marker
-		sPos.marker = c.getNextMarker(sPos, resp)
-		if c.Options.Debug {
-			if oldMarker == nil {
-				s := "nil"
-				oldMarker = &s
+		oldMarker := sPos.marker
+		newMarker := c.getNextMarker(sPos, resp)
+		if newMarker == oldMarker {
```

Comment:
I'd like a comment around what it means when `newMarker == oldMarker` and when it happens. Is it only when we've run out of content in an existing file? How does this behave for low volume databases? 

I'm mildly concerned to see us hitting the log file listing API every 5s (I think?) for 5min straight when the mysql log rolls over - concerned that might be enough to get us rate limited. It's a weird behavior there, where you just stop getting data for a while, then at around 6 minutes past the hour, you get new data and backfill the previous few minutes before carrying on as normal.

Reply:
Ah, good points. I think we can actually simplify this a bit. And can also only do the look-for-a-newer-file rigamarole if we know we're tailing postgres logs. Should eliminate issues around rate-limiting.